src/keysym.rs

```diff
@@ -4,12 +4,79 @@
 //
 // Copyright 2022 Oxide Computer Company
 
+use ascii::ToAsciiChar;
 use Keysym::*;
 
-#[derive(Debug)]
+// ascii characters have the same values as their keysym
+const ASCII_MAX: u32 = 0x7f;
+
+const KEYSYM_BACKSPACE: u32 = 0xff08;
+const KEYSYM_TAB: u32 = 0xff09;
+const KEYSYM_RETURN_ENTER: u32 = 0xff0d;
+const KEYSYM_ESCAPE: u32 = 0xff1b;
+const KEYSYM_INSERT: u32 = 0xff63;
+const KEYSYM_DELETE: u32 = 0xffff;
+const KEYSYM_HOME: u32 = 0xff50;
+const KEYSYM_END: u32 = 0xff57;
+const KEYSYM_PAGE_UP: u32 = 0xff55;
+const KEYSYM_PAGE_DOWN: u32 = 0xff56;
+const KEYSYM_PRINT: u32 = 0xff61;
+const KEYSYM_PAUSE: u32 = 0xff13;
+const KEYSYM_CAPS_LOCK: u32 = 0xffe5;
+const KEYSYM_SUPER_LEFT: u32 = 0xffeb;
+const KEYSYM_SUPER_RIGHT: u32 = 0xffec;
+const KEYSYM_MENU: u32 = 0xff67;
+
+const KEYSYM_LEFT: u32 = 0xff51;
+const KEYSYM_UP: u32 = 0xff52;
+const KEYSYM_RIGHT: u32 = 0xff53;
+const KEYSYM_DOWN: u32 = 0xff54;
+// function keys are in the range: 0xffbe to 0xffc9, in order
+const KEYSYM_F1: u32 = 0xffbe;
+const KEYSYM_F12: u32 = 0xffc9;
+const KEYSYM_SHIFT_LEFT: u32 = 0xffe1;
+const KEYSYM_SHIFT_RIGHT: u32 = 0xffe2;
+const KEYSYM_CTRL_LEFT: u32 = 0xffe3;
+const KEYSYM_CTRL_RIGHT: u32 = 0xffe4;
+
+// XXX(JPH): do we need to support meta keys?
+
+const KEYSYM_ALT_LEFT: u32 = 0xffe9;
+const KEYSYM_ALT_RIGHT: u32 = 0xffea;
+const KEYSYM_SCROLL_LOCK: u32 = 0xff14;
+const KEYSYM_NUM_LOCK: u32 = 0xff7f;
+
+const KEYSYM_KP_ENTER: u32 = 0xff8d;
+const KEYSYM_KP_SLASH: u32 = 0xffaf;
+const KEYSYM_KP_ASTERISK: u32 = 0xffaa;
+const KEYSYM_KP_MINUS: u32 = 0xffad;
+const KEYSYM_KP_PLUS: u32 = 0xffab;
+const KEYSYM_KP_7: u32 = 0xffb7;
+const KEYSYM_KP_HOME: u32 = 0xff95;
+const KEYSYM_KP_8: u32 = 0xffb8;
+const KEYSYM_KP_UP: u32 = 0xff97;
+const KEYSYM_KP_9: u32 = 0xffb9;
+const KEYSYM_KP_PGUP: u32 = 0xff9a;
+const KEYSYM_KP_4: u32 = 0xffb4;
+const KEYSYM_KP_LEFT: u32 = 0xff96;
+const KEYSYM_KP_5: u32 = 0xffb5;
+const KEYSYM_KP_EMPTY: u32 = 0xff9d;
+const KEYSYM_KP_6: u32 = 0xffb6;
+const KEYSYM_KP_RIGHT: u32 = 0xff98;
+const KEYSYM_KP_1: u32 = 0xffb1;
+const KEYSYM_KP_END: u32 = 0xff9c;
+const KEYSYM_KP_2: u32 = 0xffb2;
+const KEYSYM_KP_DOWN: u32 = 0xff99;
+const KEYSYM_KP_3: u32 = 0xffb3;
+const KEYSYM_KP_PGDOWN: u32 = 0xff9b;
+const KEYSYM_KP_0: u32 = 0xffb0;
+const KEYSYM_KP_INSERT: u32 = 0xff9e;
+const KEYSYM_KP_PERIOD: u32 = 0xffae;
+const KEYSYM_KP_DELETE: u32 = 0xff9f;
+
+#[derive(Debug, Copy, Clone)]
 pub enum Keysym {
```

Comment:
nit: rename to `KeySym` to match `KeyEvent`?

can do as separate change as that would be a breaking change

Reply:
sure, but i would definitely prefer to do that in a follow on PR!

Reply:
nvm, patrick pointed out that keysym is not yet imported into propolis (other than the outstanding PR), so i don't think this would actually be a breaking change? i'll do it in this PR.

Reply:
done in b770ee1